src/main/java/edu/cornell/kfs/tax/batch/dataaccess/impl/TransactionDetailProcessorDaoJdbcImpl.java

```diff
@@ -0,0 +1,309 @@
+import java.io.UncheckedIOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.core.api.encryption.EncryptionService;
+import org.kuali.kfs.krad.bo.Note;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.DocumentHeader;
+import org.kuali.kfs.vnd.VendorConstants.AddressTypes;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+
+import edu.cornell.kfs.sys.util.CuSqlQuery;
+import edu.cornell.kfs.sys.util.CuSqlQueryPlatformAwareDaoBaseJdbc;
+import edu.cornell.kfs.tax.CUTaxConstants;
+import edu.cornell.kfs.tax.batch.TaxBatchConfig;
+import edu.cornell.kfs.tax.batch.TaxStatistics;
+import edu.cornell.kfs.tax.batch.dataaccess.TransactionDetailHandler;
+import edu.cornell.kfs.tax.batch.dataaccess.TransactionDetailProcessorDao;
+import edu.cornell.kfs.tax.batch.dataaccess.TransactionDetailRowMapper;
+import edu.cornell.kfs.tax.batch.dataaccess.TransactionDetailRowMapperFactory;
+import edu.cornell.kfs.tax.batch.dto.DocumentHeaderLite.DocumentHeaderField;
+import edu.cornell.kfs.tax.batch.dto.NoteLite;
+import edu.cornell.kfs.tax.batch.dto.NoteLite.NoteField;
+import edu.cornell.kfs.tax.batch.dto.VendorAddressLite;
+import edu.cornell.kfs.tax.batch.dto.VendorAddressLite.VendorAddressField;
+import edu.cornell.kfs.tax.batch.dto.VendorDetailLite;
+import edu.cornell.kfs.tax.batch.dto.VendorDetailLite.VendorField;
+import edu.cornell.kfs.tax.batch.dto.VendorQueryResults;
+import edu.cornell.kfs.tax.batch.metadata.TaxDtoDbMetadata;
+import edu.cornell.kfs.tax.batch.service.TaxTableMetadataLookupService;
+import edu.cornell.kfs.tax.batch.util.TaxQueryBuilder;
+import edu.cornell.kfs.tax.batch.util.TaxQueryUtils.Criteria;
+import edu.cornell.kfs.tax.batch.util.TaxQueryUtils.QuerySort;
+import edu.cornell.kfs.tax.businessobject.TransactionDetail;
+import edu.cornell.kfs.tax.businessobject.TransactionDetail.TransactionDetailField;
+import edu.cornell.kfs.vnd.CUVendorConstants.CUAddressTypes;
+
+public class TransactionDetailProcessorDaoJdbcImpl extends CuSqlQueryPlatformAwareDaoBaseJdbc
+        implements TransactionDetailProcessorDao {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private TaxTableMetadataLookupService taxTableMetadataLookupService;
+    private EncryptionService encryptionService;
+
+    @Override
+    public <U> TaxStatistics processTransactionDetails(final TaxBatchConfig config,
+            final TransactionDetailRowMapperFactory<U> rowMapperFactory,
+            final TransactionDetailHandler<U> handler) throws SQLException {
```

Comment:
Is the throws SQLException needed? I'm okay with leaving it if it is needed once implementation happens.

Reply:
Given the exception handling of the JDBC Template, I might be able to remove the `throws SQLException` part here. I'll look into removing it.